code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -53,22 +54,24 @@
 		kill_foam()
 		return
 
-	var/fraction = 1/initial(lifetime)
+	var/fraction = 1/initial(reagent_divisor)
```

Comment:
you should sanity check the fraction is an integer and probably also that it's >= 0